simpleclient/src/main/java/io/prometheus/client/CollectorRegistry.java

```diff
@@ -125,31 +125,53 @@ public void clear() {
   public Enumeration<Collector.MetricFamilySamples> metricFamilySamples() {
     return new MetricFamilySamplesEnumeration();
   }
+
+  public Enumeration<Collector.MetricFamilySamples> metricFamilySamples(Set<String> includedNames) {
```

Comment:
> Are you referring to Prometheus server development?

No, this is all in the client libraries. This is a very new feature, and it's not fully clear how it'll develop.

> Are you saying the we might need regex matching for the names[] parameters instead of just the simple contains check I went with, or should I hold off on that?

We don't know yet, it'll be a different parameter if it happens.

> Do you want me to rename the method to filteredMetricFamilySamples?

Sounds good.